src/Team-Services-Bot.Api.UnitTests/Dialogs/ConnectDialogTests.cs

```diff
@@ -69,19 +71,21 @@ public ConnectDialogTests()
             GlobalConfiguration.Configure(config => config.DependencyResolver = new AutofacWebApiDependencyResolver(container));
             var root = new RootDialog();
 
-            var toUser = await this.Fixture.GetResponse(container, root, toBot);
+            // First trigger the welcome message.
+            await this.Fixture.GetResponse(container, root, toBot1);
+            var toUser = await this.Fixture.GetResponse(container, root, toBot2);
 
             var attachment = toUser.Attachments.FirstOrDefault();
             Assert.IsNotNull(attachment, "Expecting an attachment.");
 
-            var card = attachment.Content as SigninCard;
+            var card = attachment.Content as LogOnCard;
             Assert.IsNotNull(card, "Missing signin card.");
 
             var button = card.Buttons.FirstOrDefault();
             Assert.IsNotNull(button, "Button is missing");
 
             var expected =
-                $"https://app.vssps.visualstudio.com/oauth2/authorize?client_id={appId}&response_type=Assertion&state={toBot.ChannelId};{toBot.From.Id}&scope={scope}&redirect_uri={authorizeUrl}/";
+                $"https://app.vssps.visualstudio.com/oauth2/authorize?client_id={appId}&response_type=Assertion&state={toBot2.ChannelId};{toBot2.From.Id}&scope={scope}&redirect_uri={authorizeUrl}/";
             Assert.AreEqual(expected.ToLower(), button.Value.ToString().ToLower(), "OAuth url is invalid.");
         }
```

Comment:
I wonder if this could be done as multiple unit tests rather than have a lot of asserts in a single test? Extracting the setup code into methods with intent-revealing names will also make the test much more readable.